File: EDA/semana_1/Pruebas/c++ src/countingSort.cpp
#include <iostream>
#include "Timer.hpp"
using namespace std;
#define max 1000

void counting_sort(int* &array, int n)
{
    int *count = new int[max + 1]{0};
    for (int i = 0; i < n; i++)
        count[array[i]]++;


    for (int i = 0; i < max ; i++)
        count[i + 1] = count[i] + count[i + 1];

    for (int i = max; i > 0; i--)
        count[i] = count[i-1];
    count[0] = 0;

    int *newA = new int[n]{0};

    for (int i = 0; i < n; i++)
    {
        newA[count[array[i]]] = array[i];
        count[array[i]]++;
    }
    
    delete [] array;
    array = newA;
}

void print(int *new_array)
{
    for (int i = 0; i < max; i++)
    {
        cout << new_array[i] << " ";
    }
}
int main()
{
    int numArrays;
    cin >> numArrays;

    for (int i = 0; i < numArrays; ++i)
    {
        int n;
        cin >> n;
        int *arr = new int[n];
        for (int i = 0; i < n; ++i)
            cin >> arr[i];
        Timer timer;
        timer.start();
        counting_sort(arr, n);
        timer.stop();
        cout << n << ' ' << timer.elapsedMilliseconds() << endl;
        delete [] arr;
    }
    return 0;
}

File: EDA/semana_1/Pruebas/python src/countingSort.py
from timer import Timer
#arreglo ordenado
array = []
max = 1000

def counting_sort(arr,n):
    count = [0 for i in range(max+1)]

    for i in range(n):
        count[arr[i]] += 1
 
    for i in range(max):
        count[i+1] = count[i] + count[i+1]

    for i in range(max,0,-1):
        count[i] = count[i-1]
    count[0] = 0
    
    newA = [0 for i in range(n)]

    for i in range(n):
        newA[count[arr[i]]] = arr[i]
        count[arr[i]] += 1
    arr = newA

if __name__ == "__main__":
   
    index = 0
    numArrays = int(input())
    for i in range(numArrays):
        promedio = 0
        arr = []    
        n = int(input())
        for j in range(10):
            aux = input().split(" ")
            for k in range(n):
                arr.append(int(aux[k]))

            t=Timer(n)   
            t.start()
            counting_sort(arr,n)
            t.stop()
            promedio += t.printTime()
        print(n,round(promedio/10,3))
        
       
